packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetService.java

```diff
@@ -133,13 +136,34 @@ public void deleteAsset(final UUID id) throws IOException {
 	 * @throws IOException If there is an error creating the asset
 	 */
 	public T createAsset(final T asset) throws IOException {
-		final UUID id = UUID.randomUUID();
-		asset.setId(id);
+		if (elasticService.documentExists(getAssetIndex(), asset.getId().toString())) {
+			throw new ResponseStatusException(
+					HttpStatus.CONFLICT, "Asset already exists with ID: " + asset.getId());
+		}
 		asset.setCreatedOn(Timestamp.from(Instant.now()));
 		elasticService.index(getAssetIndex(), asset.getId().toString(), asset);
 		return asset;
 	}
 
+	/**
+	 * Create new assets and saves to ES
+	 *
+	 * @param asset The asset to create
+	 * @return The created asset
+	 * @throws IOException If there is an error creating the asset
+	 */
+	public List<T> createAssets(final List<T> assets) throws IOException {
+		for (final T asset : assets) {
+			if (elasticService.documentExists(getAssetIndex(), asset.getId().toString())) {
+				throw new ResponseStatusException(
```

Comment:
Because this isn't a controller, do we want to throw a different kind of exception here to use it more generically? (Then let the controller package it nicely?)

Reply:
Done!